crypto/dh/dh_ameth.c

```diff
@@ -536,6 +536,60 @@ static int dhx_pkey_import_from(const OSSL_PARAM params[], void *vpctx)
     return dh_pkey_import_from_type(params, vpctx, EVP_PKEY_DHX);
 }
 
+#define dh_bn_dup_check(bn) \
+{ \
+  const BIGNUM *f = DH_get0_##bn(dh); \
+  \
+  if (f != NULL && (dupkey->bn = BN_dup(f)) == NULL) \
+      goto err; \
+}
+
+static DH *dh_dup(const DH *dh)
+{
+    DH *dupkey = NULL;
+
+    /* Do not try to duplicate foreign DH keys */
+    if (ossl_dh_get_method(dh) != DH_OpenSSL())
+        return NULL;
+
+    if ((dupkey = ossl_dh_new_ex(dh->libctx)) == NULL)
+        return NULL;
+
+    dupkey->length = DH_get_length(dh);
+    if (!ossl_ffc_params_copy(&dupkey->params, &dh->params))
+        goto err;
+
+    dh_bn_dup_check(pub_key)
+    dh_bn_dup_check(priv_key)
+    if (!CRYPTO_dup_ex_data(CRYPTO_EX_INDEX_DH,
```

Comment:
This is not currently defined in the fips module

Reply:
EVP_PKEY_dup is not included in FIPS module. I do not think there is a reason to include it there ever. dh_ameth.c is not included in FIPS module either.